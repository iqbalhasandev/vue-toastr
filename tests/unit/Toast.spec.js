import { shallowMount } from '@vue/test-utils';
import Toast from '@/components/Toast.vue';

const mockProps = {
	data: {
		msg: 'Toast Msg',
		progressbar: false,
		timeout: 1000,
		title: 'Toast Title',
		type: 'error'
	}
};
describe('Toast.vue', () => {
	// it("default props value", () => {
	//   const wrapper = shallowMount(Toast, {});
	//   expect(wrapper.props()).toEqual(mockProps);
	// });

	it('renders props when passed', () => {
		const wrapper = shallowMount(Toast, {
			propsData: mockProps
		});
		expect(wrapper.props()).toEqual(mockProps);
	});

	it('match attributes', () => {
		const wrapper = shallowMount(Toast, {
			propsData: mockProps
		});
		expect(wrapper.attributes()).toEqual({
			class: 'toast toast-error',
			style: 'display: block;'
		});
	});
	it('match classNames', () => {
		const props = {
			data: { ...mockProps.data, classNames: [ 'animated', 'bounce' ] }
		};
		const wrapper = shallowMount(Toast, {
			propsData: props
		});
		expect(wrapper.attributes()).toEqual({
			class: 'toast toast-error animated bounce',
			style: 'display: block;'
		});
	});
	it('matches slot msg', () => {
		const wrapper = shallowMount(Toast, {
			propsData: mockProps,
			slots: {
				default: '<div class="fake-msg"></div>'
			}
		});
		expect(wrapper.contains('.fake-msg')).toBe(true);
	});

	it('matches snapshot', () => {
		const wrapper = shallowMount(Toast, {
			propsData: mockProps
		});
		expect(wrapper.html()).toMatchSnapshot();
	});
});
