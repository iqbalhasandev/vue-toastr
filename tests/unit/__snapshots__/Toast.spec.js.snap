// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast.vue matches snapshot 1`] = `
<div class="toast toast-error" style="display: block;">
  <!---->
  <div class="toast-title">Toast Title</div>
  <div class="toast-message">Toast Msg</div>
  <!---->
</div>
`;
